drivers/frequency/adf5902/adf5902.h

```diff
@@ -0,0 +1,585 @@
+#define ADF5902_REG14_DEV_WORD(x)	(((x) & 0xFFFF) << 5)
+#define ADF5902_REG14_DEV_OFFSET(x)	(((x) & 0xF) << 21)
+#define ADF5902_REG14_DEV_SEL(x)	(((x) & 0x3) << 25)
+#define ADF5902_REG14_TX_RAMP_CLK(x)	(((x) & 0x1) << 30)
+#define ADF5902_REG14_TX_DATA_INV(x)	(((x) & 0x1) << 31)
+#define ADF5902_REG14_RESERVED		(0x0 << 27)
+
+/* Register 14 Bit Definitions */
+#define ADF5902_MAX_DEV_WORD		0x7FFF
+#define ADF5902_MIN_DEV_WORD		(int16_t)0x8000
+
+#define ADF5902_MAX_DEV_OFFSET		0x9
+#define ADF5902_MIN_DEV_OFFSET		0x0
+
+#define ADF5902_DEV_SEL_0		0x0
+#define ADF5902_DEV_SEL_1		0x1
+#define ADF5902_DEV_SEL_2		0x2
+#define ADF5902_DEV_SEL_3		0x3
+
+#define ADF5902_TX_RAMP_CLK_DIV		0x0
+#define ADF5902_TX_RAMP_TX_DATA_PIN	0x1
+
+#define AD5902_TX_DATA_INV_DISABLE	0x0
+#define AD5902_TX_DATA_INV_ENABLE	0x1
+
+/* Register 15 Map */
+#define ADF5902_REG15_STEP_WORD(x)	(((x) & 0xFFFFF) << 5)
+#define ADF5902_REG15_STEP_SEL(x)	(((x) & 0x3) << 25)
+#define ADF5902_REG15_RESERVED		(0x0 << 27)
+
+/* Register 15 Bit Definition */
+#define ADF5902_MIN_STEP_WORD		0x00000
+#define ADF5902_MAX_STEP_WORD		0xFFFFF
+
+#define ADF5902_STEP_SEL_0		0x0
+#define ADF5902_STEP_SEL_1		0x1
+#define ADF5902_STEP_SEL_2		0x2
+#define ADF5902_STEP_SEL_3		0x3
+
+/* Register 16 Map */
+#define ADF5902_REG16_DEL_START_WORD(x)	(((x) & 0xFFF) << 5)
+#define ADF5902_REG16_RAMP_DEL(x)	(((x) & 0x1) << 19)
+#define ADF5902_REG16_TX_DATA_TRIG(x)	(((x) & 0x1) << 20)
+#define ADF5902_REG16_DEL_SEL(x)	(((x) & 0x3) << 23)
+#define ADF5902_REG16_RESERVED		((0x0 << 17) | (0x0 << 21) | (0x1 << 25))
+
+#define ADF5902_MIN_DELAY_START_WRD	0x000
+#define ADF5902_MAX_DELAY_START_WRD	0xFFF
+
+#define ADF5902_RAMP_DEL_DISABLE	0x0
+#define ADF5902_RAMP_DEL_ENABLE		0x1
+
+#define ADF5902_TX_DATA_TRIG_DISABLE	0x0
+#define ADF5902_TX_DATA_TRIG_ENABLE	0x1
+
+#define ADF5902_DEL_SEL_0		0x0
+#define ADF5902_DEL_SEL_1		0x1
+#define ADF5902_DEL_SEL_2		0x2
+#define ADF5902_DEL_SEL_3		0x3
+
+/* Register 17 Map */
+#define ADF5902_REG17_RESERVED		(0x0 << 5)
+
+/** Specifications */
+#define ADF5902_MAX_VCO_FREQ		24250000000ull
+#define ADF5902_MIN_VCO_FREQ		24000000000ull
+#define ADF5902_MIN_REFIN_FREQ		10000000
+#define ADF5902_MAX_REFIN_FREQ		260000000
+#define ADF5902_MAX_FREQ_PFD		110000000
+#define ADF5902_MAX_SLOPE_NO		4
+#define ADF5902_MAX_DELAY_WORD_NO	4
+#define ADF5902_MAX_CLK2_DIV_NO		4
+#define ADF5902_VLSB				0.00733f
+#define ADF5902_VOFF				0.699f
+#define ADF5902_VGAIN				0.0064f
+#define ADF5902_SPI_DUMMY_DATA		0x0
+#define ADF5902_FREQ_CAL_DIV_100KHZ	100000
+#define ADF5902_CLK1_DIV_25KHZ		25000
+#define ADF5902_ADC_CLK_DIV_1MHZ	1000000
+#define ADF5902_BUFF_SIZE_BYTES		4
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+struct slope {
+	/* Deviation Word */
+	int16_t dev_word;
+
```

Comment:
nitpick: seems like extra line here.

Reply:
I added extra line between each parameter for a better readability. But I can remove them all.

Reply:
Removed.